src/tasks/WasmAppBuilder/WasmAppBuilder.cs

```diff
@@ -187,9 +188,22 @@ private bool ExecuteInternal ()
         var asmRootPath = Path.Combine(AppDir, config.AssemblyRootFolder);
         Directory.CreateDirectory(AppDir!);
         Directory.CreateDirectory(asmRootPath);
+        if (UseWebcil)
+            Log.LogMessage (MessageImportance.Normal, "Converting assemblies to Webcil");
         foreach (var assembly in _assemblies)
         {
-            FileCopyChecked(assembly, Path.Combine(asmRootPath, Path.GetFileName(assembly)), "Assemblies");
+            if (UseWebcil)
+            {
+                var tmpWebcil = Path.GetTempFileName();
+                var webcilWriter = Microsoft.WebAssembly.Build.Tasks.WebcilConverter.FromPortableExecutable(inputPath: assembly, outputPath: tmpWebcil, logger: Log);
+                webcilWriter.ConvertToWebcil();
+                var finalWebcil = Path.ChangeExtension(assembly, ".webcil");
+                FileCopyChecked(tmpWebcil, Path.Combine(asmRootPath, Path.GetFileName(finalWebcil)), "Assemblies");
```

Comment:
is there a reason you aren't doing something like
```suggestion
                var finalWebcil = Path.Combine(asmRootPath, Path.GetFileName(Path.ChangeExtension(assembly, ".webcil")), "Assemblies");
                var webcilWriter = Microsoft.WebAssembly.Build.Tasks.WebcilConverter.FromPortableExecutable(inputPath: assembly, outputPath: finalWebcil, logger: Log);
                webcilWriter.ConvertToWebcil();
                _fileWrites.Add(finalWebcil);
```

Reply:
@lewing I think I misunderstood `FileCopyChecked`.  I thought it was avoiding writing to `dst` if it already exists with the same content.  My intention was to avoid messing with modification times of `finalWebcil` unless something really changed.

Reply:
You can use `Utils.CopyIfDifferent(string src, string dst, bool useHash)`.

Reply:
https://github.com/dotnet/runtime/blob/0e12347861940389153ee2233ff398d551f86711/src/tasks/WasmAppBuilder/EmccCompile.cs#L216-L219